website/api_versioned_docs/version-0.19/zkvm/tutorials/hello-world.md

```diff
@@ -0,0 +1,111 @@
+}
+```
+
+## Step 3 (Guest): Read input and commit output
+
+Now, let's look at the guest code located in `methods/guest/src/main.rs`.
+This is the portion of the code that will be proven.
+In the code snippet below, the guest reads the `input` value from the host and then commits it to the [journal] portion of the [receipt].
+
+```rust ignore
+use risc0_zkvm::guest::env;
+
+pub fn main() {
+    // read the input
+    let input: u32 = env::read();
+
+    // do something with the input
+    // write public output to the journal
+    env::commit(&input);
+}
+```
+
+The `env::commit` function commits public results to the [journal]. Once committed to the journal, anyone with the [receipt] can read this value.
+
+Notice, by committing any private information to the journal, we make this private data public. We want to avoid committing sensitive data to public journal.
+
+## Step 4 (Host): Generate a receipt and read its journal contents
+
+Let's look at how the host generates a receipt and extracts the [journal]'s contents.
+We get a receipt, extract a journal from the receipt, and verify it.
+In a real-world scenario, we'd want to hand the [receipt] to someone else, but verifying it ourselves will be a nice way to check our project is working as expected.
+After receipt verification, let's print `Hello world` with the public output by adding this line to the host: `println!("Hello, world! I generated a proof of guest execution! {} is a public output from journal", _output);`
+
+```rust ignore
+use methods::{
+    HELLO_GUEST_ELF, HELLO_GUEST_ID
+};
+use risc0_zkvm::{default_prover, ExecutorEnv};
+
+fn main() {
+    let input: u32 = 15 * u32::pow(2, 27) + 1;
+    let env = ExecutorEnv::builder().write(&input).unwrap().build().unwrap();
+    // Obtain the default prover.
+    let prover = default_prover();
+    // Produce a receipt by proving the specified ELF binary.
+    let receipt = prover.prove_elf(env, HELLO_GUEST_ELF).unwrap();
+
+    // Extract journal of receipt
+    let _output: u32 = receipt.journal.decode().unwrap();
+
+    receipt.verify(HELLO_GUEST_ID).unwrap();
+
+    // Print, notice, after committing to a journal, the private input became public
+    println!("Hello, world! I generated a proof of guest execution! {} is a public output from journal ", _output);
+}
+```
+
+You should now be able to see your proof with `cargo run --release`.
+If your program printed the "Hello, world!" assertion and [receipt] verification was a success, congratulations!
+If not, we hope that troubleshooting will get you familiar with the system, and we'd love to chat with you on [Discord].
+Or, if you believe you've found a bug or other problem in our code, please open an [issue] describing the problem.
+
+If you're ready to start building more complex projects, we recommend taking a look at the other examples in our [examples directory] for more project ideas that use zero-knowledge proofs.
+
+[Installation]: /zkvm/install
```

Comment:
```suggestion
[Installation]: /../../install
```

Reply:
`/../install `

Reply:
The important thing is that we use relative paths rather than absolute ones. By putting a `/` at the start, this makes it absolute.